gnucash/gnucash-commands.cpp

```diff
@@ -434,3 +547,137 @@ Gnucash::report_list (void)
+        scm_c_primitive_load (args->script->c_str());
+    }
+    if (args->interactive)
+    {
+        std::cout << _("Welcome to Gnucash Interactive Guile Session") << std::endl;
+        std::vector<const char*> modules =
+            { "gnucash core-utils", "gnucash engine", "gnucash app-utils", "gnucash report",
+              "system repl repl", "ice-9 readline" };
+        auto show_and_load = [](const auto& mod)
+        {
+            std::cout << bl::format ("(use-modules ({1}))") % mod << std::endl;
+            scm_c_use_module (mod);
+        };
+        std::for_each (modules.begin(), modules.end(), show_and_load);
+        scm_c_eval_string ("(activate-readline)");
+        scm_c_eval_string ("(start-repl)");
+    }
+    cleanup_and_exit_with_save (args);
+}
+
+
+int
+Gnucash::run_scripting (std::vector<std::string> m_script_args,
+                        const bo_str& m_file_to_load,
+                        std::string& m_language,
+                        const bo_str& m_script,
+                        bool m_open_readwrite,
+                        bool m_interactive)
+{
+    std::vector<std::string> errors;
+    static const std::vector<std::string> languages = { "guile", "python" };
+
+    if (m_open_readwrite && !m_file_to_load)
+        errors.push_back (_ ("--readwrite: missing datafile!"));
+
+    if (m_script && (!boost::filesystem::is_regular_file (*m_script)))
+        errors.push_back ((bl::format (_("--script: {1} is not a file")) % *m_script).str());
+
+    if (std::none_of (languages.begin(), languages.end(),
+                      [&m_language](auto& lang){ return m_language == lang; }))
+        errors.push_back (_ ("--language: must be 'python' or 'guile'"));
+#ifndef HAVE_PYTHON_H
+    else if (m_language == "python")
+        errors.push_back (_("--language: python wasn't compiled in this build"));
+#endif
+
+    if (!errors.empty())
+    {
+        std::cerr << _("Errors parsing arguments:") << std::endl;
+        auto to_console = [](const auto& str){ std::cerr << str << std::endl; };
+        std::for_each (errors.begin(), errors.end(), to_console);
+        gnc_shutdown_cli (1);
+    }
+
+    std::vector<const char*> newArgv {"gnucash-cli"};
+    std::transform (m_script_args.begin(), m_script_args.end(), std::back_inserter(newArgv),
+                    [](const std::string& s) { return s.c_str(); });
+    // note the vector<const char*> is valid as long as m_script_args's strings are not damaged!
+
+    gnc_prefs_init ();
+    gnc_ui_util_init();
+    if (m_file_to_load)
+        load_file (*m_file_to_load, m_open_readwrite);
+
+    scripting_args args { m_script, m_language, m_interactive };
+    if (m_language == "guile")
+    {
+        scm_boot_guile (newArgv.size(), (char**)newArgv.data(), run_guile_cli, &args);
+        return 0;               // never reached...
+    }
+
+#ifdef HAVE_PYTHON_H
+    else if (m_language == "python")
```

Comment:
The following block should be a separate function `run_python_cli` mirroring `run_guile_cli`.